Refactor api OpenAI

YasserSB opened the pull request:
https://jam.dev/c/e471bebb-b3fe-4afd-a4e2-7ab330a95e00

Commit:
feat(openai): add base url configuration and utility functions

- Add OPENAI_BASE_URL to environment variables and app config
- Create new openaiConfig.ts with utility functions for API access
- Implement fallback for base URL and error handling for API key

Commit:
refactor(openai): centralize openai configuration and endpoints

Move OpenAI API key retrieval and endpoint configuration to a dedicated module.
Update all files to use the centralized configuration instead of direct imports.
This improves maintainability and makes API configuration changes easier.

Commit:
fix(transactionAnalyzer): update OpenAI model name to correct version

randhipp commented on .env.example:
```diff
@@ -1,3 +1,4 @@
 OPENAI_API_KEY=your_api_key
+OPENAI_BASE_URL=https://api.openai.com
```
should we differentiate the url and api key for sumopod
```
SUMOPOD_AI_API_KEY=
SUMOPOD_AI_BASE_URL=
```
so we can correctly fallback to old API with the correct API_KEY?

YasserSB replied:
Got it mas

randhipp commented:
@YasserSB can you please add screenshot / demo that the new API working?

Commit:
feat: add SumoPod AI configuration support

Add new environment variables and configuration for SumoPod AI integration as an alternative to OpenAI. Update the OpenAI config utility to fallback to SumoPod AI settings when available.

Commit:
fix(openaiConfig): add fallback URL and error handling for OpenAI config

Add a default fallback URL and error handling when neither sumopodAiBaseUrl nor openaiBaseUrl is found in the config. Also log the base URL being used for debugging purposes.

randhipp commented:
<img width="1310" height="671" alt="Screenshot 2025-07-15 at 11 21 53" src="https://github.com/user-attachments/assets/f516db95-5260-4467-a951-f9981f3cbdff" />

success using sumopod API, checked on local with current PR branch using 
```
npx expo run:android
```
